scatter_gather.go

```diff
@@ -29,21 +29,22 @@ func (sg *scatterGather[T, R]) scatter(ctx context.Context, forEach func(T) (<-c
 			default:
 			}
 
-			sout, err := forEach(target)
+			cctx, cncl := context.WithTimeout(ctx, sg.maxWait)
+			sout, err := forEach(cctx, target)
+			cncl()
 			if err != nil {
 				log.Errorw("failed to scatter on target", "target", target, "err", err)
 				return
 			}
 
-			for {
-				select {
-				case <-ctx.Done():
-					return
-				case r, ok := <-sout:
-					if !ok {
+			if sout != nil {
+				if ctx.Err() == nil {
+					select {
+					case <-ctx.Done():
+						return
+					case sg.out <- *sout:
```

Comment:
This change assumes the channel returned by `forEach` will/can only have one element. We probably should document this in `scatter` godoc.

Reply:
i simplified so that forEach returns directly - it's already in a go routine, so having 2 layers of channels for each scattered element was starting to feel overly complicated